cmake/test-targets.cmake

```
@@ -65,6 +65,17 @@ else ()
     message(STATUS "Set PERLCRITIC_PATH to the path of the perlcritic executable to enable Perl syntax/style checks.")
 endif ()
 
+# add test for bash script syntax
+find_program(SH_PATH shfmt)
+if (SH_PATH)
+    add_test(
+        NAME test-local-bash-syntax
+        COMMAND sh -c "${CMAKE_CURRENT_SOURCE_DIR}/tools/check-bash-scripts ${CMAKE_SOURCE_DIR}"
+    )
```

Please make actually use of `SH_PATH` so users can specify a different path by invoking CMake like `-DSH_PATH=/custom/shfmt`:

```suggestion
    add_test(
        NAME test-local-bash-syntax
        COMMAND "${CMAKE_CURRENT_SOURCE_DIR}/tools/check-bash-scripts" "${SH_PATH}" "${SHELLCHECK_PATH}"
        WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
    )
```

Note that `${CMAKE_SOURCE_DIR}` also lacked quoting on shell-level and it is generally a bad practice to assume `sh` exists in CMake (for the sake of portability). So one would usually use https://cmake.org/cmake/help/latest/manual/cmake.1.html#run-a-command-line-tool instead (example: https://github.com/Martchus/syncthingtray/blob/b55480548752fb77f5d244f7ab4dc34cd7d5d9a2/syncthing/CMakeLists.txt#L185). This also allows passing environment variables which would be an alternative way of passing the executable's path if you don't like using the first argument for that.

---

Why would you pass `"${SH_PATH}" "${SHELLCHECK_PATH}"` here?
The shellcheck target is some lines above...

---

That was a C&P mistake. I amended my suggestion.